includes/helper/class-wp-job-manager-com-auth-token.php

```diff
@@ -0,0 +1,179 @@
+	 */
+	private function __construct() {}
+
+	/**
+	 * Returns the singleton instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Generates the site token associated with an object type and object id
+	 *
+	 * @param string $object_type Type of the object associated with the token. Accepts 'post' or 'user'.
+	 * @param int    $object_id The ID of the object to associate the token with.
+	 * @return string|WP_Error The token generated, or a WP_Error if the token could not be generated/persisted
+	 */
+	public function generate( $object_type, $object_id ) {
+		if ( ! in_array( $object_type, self::ACCEPTED_OBJECT_TYPES, true ) ) {
+			return new WP_Error( 'wpjobmanager-com-invalid-type', __( 'Invalid object type', 'wp-job-manager' ) );
+		}
+		$token = $this->generate_new_token();
+		if ( is_wp_error( $token ) ) {
+			return $token;
+		}
+		$encoded = $this->encode( $token );
+		$result  = add_metadata( $object_type, $object_id, self::META_KEY, $encoded );
+		if ( ! $result ) {
+			return new WP_Error( 'wpjobmanager-com-token-not-saved', __( 'Token could not be persisted', 'wp-job-manager' ) );
+		}
+		return $token;
+	}
+
+	/**
+	 * Generates a new random token and return it
+	 *
+	 * @return string|WP_Error The random token generated, or a WP_Error if the token could not be generated.
+	 */
+	private function generate_new_token() {
+		try {
+			$hash = random_bytes( 48 );
+			//phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+			$base64 = base64_encode( $hash );
+			return str_replace( [ '+', '/', '=' ], '', $base64 );
+		} catch ( Exception $e ) {
+			return new WP_Error( 'wpjobmanager-com-token-not-generated', __( 'Token could not be generated', 'wp-job-manager' ) );
+		}
+	}
+
+	/**
+	 * Prepare the token to be persisted in the database
+	 *
+	 * @param string $token The token to encode.
+	 * @return array The value to persist in the database.
+	 */
+	private function encode( $token ) {
+		return [
+			'token' => wp_hash_password( $token ),
+			'ts'    => time(),
+		];
+	}
+
+	/**
+	 * Validate if a token is valid or not, and might remove it if it is valid or expired.
+	 *
+	 * @param string $object_type Type of object metadata is for. Accepts 'post' or 'user'.
+	 * @param int    $object_id  The ID of the object to associate the token with.
+	 * @param string $token The token to validate.
+	 * @return bool True if the token is valid, false otherwise.
+	 */
+	public function validate( $object_type, $object_id, $token ) {
+		if ( ! in_array( $object_type, self::ACCEPTED_OBJECT_TYPES, true ) ) {
+			return false;
+		}
+		$metadatas = get_metadata( $object_type, $object_id, self::META_KEY );
+		if ( false === $metadatas ) {
+			return false;
+		}
+		foreach ( $metadatas as $metadata ) {
+			if ( ! $this->is_valid( $metadata ) ) {
```

Comment:
WDYT to rename it to `is_valid_format` or something similar? It seems the main validation method.

Reply:
Fair, renamed in 5f0ea4b519579ce82e49ddb2c9abb50dee44cfb4. Thanks!